eZ/Publish/Core/Persistence/Cache/Readme.md

```diff
@@ -0,0 +1,72 @@
+# CORE DEV DOC: Persistence\Cache
+
+SPI Persistence Cache is a layer aiming to cache calls to Persistence (backend database) deemed costly under load. It
+tries to balance this need up against the complexity and additional [system overhead](#guidelines-for-core-development) imposed by caching.
+
+## Design
+
+Handlers using `AbstractHandler` internally consist of one cache layer:
+
+- "Shared cache": A Symfony Cache based cache pool, supporting a range of cache adapters like filesystem, Redis, Memcached.
+   Note: Due to being shared by design, clusters need remote cache, thus multi lookups are strongly advice to reduce round trip latency.
+
+Handlers using `AbstractInMemoryHandler` / `AbstractInMemoryPersistenceHandler` in addition adds a second cache layer in front of "Shared cache":
+
+- "InMemory cache": A burst cache in PHP aiming at covering the most heavily used parts of the Content model to reduce repeated lookups to remote cache system.
+   Note: It's not shared but per request/process. To keep risk of race condition negligible, it has own milliseconds ttl & item limits.
+
+There are abstract test classes for the respective abstract classes above, these are opinionated and enforce conventions to:
+- Avoid too much logic in cache logic _(e.g. warm-up logic)_, which can be a source of bugs.
+- Avoids having to write error prone test cases for each and every method.
+
+_This makes sure cache layer is far less complex to maintain and evolve then what was the case in 1.x._
+
+
+### Tags
+
+List of content tags that can be somewhat safely semi "officially" used to clear related entities in cache:
+- `content-<id>`: Cache tag which refers to Content/ContentInfo entity.
+- `location-<id>`: Cache tag which refers to Locations and/or their assigned Content/ContentInfo entities.
+- `location-path-<id>`: Like tag above but applied to all Content/Locations in the subtree of this ID, so can be used by tree operations.
```

Comment:
```suggestion
- `location-path-<id>`: Like the tag above but applied to all Content/Locations in the subtree of this ID, so it can be used by tree operations.
```